packages/next/client/image.tsx

```diff
@@ -99,8 +99,29 @@ type CallLoaderProps = {
   width: number
   quality?: number
 }
+type URLResolver = (resolverProps: CallLoaderProps) => string
+
+let customResolver: URLResolver
+export function registerCustomResolver(resolver: URLResolver): void {
+  if (process.env.NODE_ENV !== 'production' && configLoader !== 'custom') {
+    throw new Error(
+      `registerCustomResolver can only be used if image loader is set to 'custom' in next.config.js`
+    )
+  }
+  customResolver = resolver
+}
 
 function callLoader(loaderProps: CallLoaderProps) {
+  if (configLoader === 'custom') {
+    if (process.env.NODE_ENV !== 'production' && configLoader !== 'custom') {
```

Comment:
This will never eval to true because the nested if statements are expecting opposite values for `configLoader`

Reply:
Ah, good point. I hadn't added tests for those error messages, just because I wasn't sure it was worth adding two additional builds to the tests just to hit those two error messages, but I guess it's necessary after all. I'll add those now.